index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
While that sounds like a noble goal, it also sounds like a problem.  To me the whole reason for introducing a `Maybe` is to make an entire class of problems go away.  If it then introduces its own potential problems, I think maybe you're looking for too much out of the elegant ideas.  It might be better to express `filter` more directly.

(A correction to my last message.  That wasn't a lawful Monoid.  I think it would have to tread `Nothing` as the id value, as does the "standard" Maybe Monoid.  Thus, `Nothing(concat(m); //=> m` and `m.concat(Nothing); //=> m`.  Otherwise we'd have to try to make an id out of `Just([])`, which would work on `Just(list)` but not `Just(scalar)`. )